exonum/src/api/node/private/mod.rs

```diff
@@ -192,11 +194,30 @@ impl SystemApi {
 
     fn handle_shutdown(self, name: &'static str, api_scope: &mut ApiScope) -> Self {
         let self_ = self.clone();
-        api_scope.endpoint_mut(name, move |_query: ()| -> Result<(), ApiError> {
+
+        let handler = move || -> Result<(), ApiError> {
             self.sender
                 .send_external_message(ExternalMessage::Shutdown)
                 .map_err(ApiError::from)
-        });
+        };
+
+        let index = move |_request: HttpRequest| -> BackendFutureResponse {
+            let handler = handler.clone();
+
+            let future = Ok(handler())
+                .and_then(|_| Ok(HttpResponse::Ok().json(())))
```

Comment:
Can we return `Ok(HttpResponse::Ok().json(()))` in the `handler` and get rid of this `and_then`?

IMHO It'd be better in the form of the

```rust
let handler = handler.clone();
let future = handler().into_future();
Box::new(future)
```

And btw I'd split it into several lines to avoid this parentheses chaos:
```rust
let ok_response = HttpResponse::Ok().json(());
Ok(ok_response)
```